main.c

```diff
@@ -119,7 +119,7 @@ union ResourceData {
     char *name;
   } name_server_record;
   struct {
-    char name;
+    char* name;
```

Comment:
I missed that one :)

Reply:
OK, I will create a new project that implements all the features.

Thank you very much.